hazelcast-enterprise/Dockerfile

```diff
@@ -1,14 +1,13 @@
 FROM redhat/ubi9-minimal:9.3
 
-# Versions of Hazelcast
-ARG HZ_VERSION=5.5.0-SNAPSHOT
-# Variant - empty for full, "slim" for slim
-ARG HZ_VARIANT=""
+# Used for image metadata only
+ARG HZ_VERSION=
```

Comment:
The only place _I think_ this is used is the metadata when published on the registry (or the openshift equivalent).

For any release job we override this with a proper value anyway, so setting a default value is redundant.

I struggled with the hardcoded version, because from the `Dockerfile` alone it's not really possible to reason what version of Hazelcast it's actually got - and the default is now the _latest released_ version which is `5.4`.

Reply:
Discussed with @ldziedziul & @nishaatr:

[Update](https://github.com/hazelcast/hazelcast-pipeline-library/blob/master/src/com/hazelcast/qe/pipeline/distribution/dockerfiles/DockerfileUpdateStep.groovy) is safe because it:
- reads the file
- does a regex replacement
- writes the string

No replacements will cause no issue there.

A [change is required to allow an empty commit when the change is pushed back](https://github.com/hazelcast/hazelcast-pipeline-library/blob/master/src/com/hazelcast/qe/pipeline/distribution/dockerfiles/CommitAndTagDocker.groovy#L31), however - @nishaatr to action.